generative_data_prep/data_prep/pipeline.py

```diff
@@ -269,8 +275,31 @@ def multiprocess_data_prep(
             )
         )
 
-    with Pool(num_workers) as p:
-        _ = p.starmap(data_prep_main, data_prep_main_args_list)
+    futures = []
+    with concurrent.futures.ProcessPoolExecutor(max_workers=num_workers) as executor:
+        futures = [executor.submit(data_prep_main_helper, args) for args in data_prep_main_args_list]
+
+    # if one process fails with an exception such as a RuntimeError, all other processes will fail with a
+    # BrokenProcessPool exception.  In such a situation we only want to show the user the RuntimeError.  We only want
+    # to show the user the BrokenProcessPool exception if all failing processes failed with this error.  This
+    # is why we have this complicated logic below.
+    broken_process_indices = []
+    broken_process_pool_exc = None
+    # search for any "interesting" exception (a non-BrokenProcessPool Exception)
+    for i, future in enumerate(futures):
+        try:
+            future.result()
+        except Exception as exc:
+            if isinstance(exc, concurrent.futures.process.BrokenProcessPool):
+                broken_process_indices.append(str(i))
+                broken_process_pool_exc = exc
+            else:
+                print(f'\n\nProcess {i} failed with the following exception:')
```

Comment:
Maybe lets give a hint about how to fix this error? Say a processes was killed by OS it is likely due to OOM, so decrease number of workers you are running or something?

Reply:
Hmm as you mentioned, OOM is just one of the possible RuntimeErrors, should I say something like if your RuntimeError is a MemoryError then decrease number of workers?